src/java.base/share/classes/java/util/concurrent/atomic/AtomicIntegerFieldUpdater.java

```diff
@@ -402,6 +402,9 @@ private static final class AtomicIntegerFieldUpdaterImpl<T>
             if (!Modifier.isVolatile(modifiers))
                 throw new IllegalArgumentException("Must be volatile type");
 
+            if (Modifier.isStatic(modifiers))
```

Comment:
Can we break out the Modifier tests in the three updater classes to a common method?

Reply:
Created https://bugs.openjdk.org/browse/JDK-8364544 to track that.